src/util/http_utils.js

```diff
@@ -664,6 +665,82 @@ function set_amz_headers(req, res) {
     res.setHeader('x-amz-id-2', req.request_id);
 }
 
+const s3_error_options = {
+    ErrorClass: S3Error,
+    error_missing_content_length: S3Error.MissingContentLength
+};
+/**
+ * @param {Object} req
+ * @param {http.ServerResponse} res
+ */
+async function set_expiration_header(req, res) {
+    if (req.method === 'HEAD' || req.method === 'GET' || req.method === 'PUT') {
+        const rules = req.params.bucket && await req.object_sdk.read_bucket_lifecycle_config_info(req.params.bucket);
+        const object_md = {
+                bucket: req.params.bucket,
+                key: req.params.key,
+                size: req.headers['x-amz-decoded-content-length'] || req.headers['content-length'] ? parse_content_length(req, s3_error_options) : undefined,
+                tagging: req.body && req.body.Tagging ? s3_utils.parse_body_tagging_xml(req) : undefined,
+        };
+
+        if (object_md.key && rules?.length > 0) { // updating x-amz-expiration if object key is present
+            for (const rule of rules) {
+                if (rule?.status !== 'Enabled') continue;
+
+                const filter = rule?.filter || {};
+
+                if (filter.prefix && !object_md?.key.startsWith(filter.prefix)) continue;
+
+                if (filter.object_size_greater_than && object_md?.size <= filter.object_size_greater_than) continue;
+                if (filter.object_size_less_than && object_md?.size >= filter.object_size_less_than) continue;
```

Comment:
Why do we continue on the `filter`?
Could you add a comment about it?

Reply:
We will move to the next rule in the `for` loop if object is not matching the `rule.filter` condition. If all the conditions are satisfied then only we set the header.